CMakeLists.txt

```diff
@@ -2,10 +2,10 @@
 # Licensed under the Apache 2.0 License.
 cmake_minimum_required(VERSION 3.16)
 
-set(ALLOWED_TARGETS "snp;virtual")
+set(ALLOWED_TEST_PLATFORMS "snp;virtual")
 
-set(COMPILE_TARGET
-    "snp"
+set(CCF_TEST_PLATFORM
+    "virtual"
```

Comment:
I argue that although this diff is small, this is fundamentally a _new_ arg, so I'm not changing a default. We're not making things less secure (the default build still produces SNP-capable binaries, that default to using SNP logic), but our _tests_ default to using the "virtual" attestations unless told otherwise. As a result, we don't have to pass `-DCOMPILE_TARGET=virtual` (or `-DCCF_TEST_PLATFORM=virtual`) in all of our CI yamls and whenever we're telling people how to build, the default is the majority-option and the few things that want to explicitly try SNP-networks can do so.

Reply:
I think the tests should detect the platform the same way the binaries do, and enforce their opinion of the platform when they act as the consortium that agrees to open the network. Telling them to do otherwise should require an override. Let's discuss.